### ISSUE TITLE
Check for Algorithm Specific Data Files Upfront

### ISSUE THREAD

**Issue by daniel-j-h** [Project-OSRM/osrm-backend#4122]
When running the new pipeline

    osrm-extract -p ../profiles/car.lua map.pbf
    osrm-partition map.osrm
    osrm-customize map.osrm
    osrm-routed map.osrm

but forgetting to specify the mld algorithm (via `osrm-routed --algorithm MLD map.osrm`) we start up the engine anyway only showing a warning about missing hsgr and core files as we always do in the mld pipeline.

We then crash hard internally on queries e.g. with

```
[assert][140497935607744] /tmp/osrm-backend/include/util/static_graph.hpp:144
in: osrm::util::StaticGraph<EdgeDataT, Ownership>::StaticGraph(osrm::util::StaticGraph<EdgeDataT, Ownership>::Vector<osrm::util::static_graph_details::NodeArrayEntry>, osrm::util::StaticGraph<EdgeDataT, Ownership>::Vector<osrm::util::static_graph_details::EdgeArrayEntry<EdgeDataT> >) [with EdgeDataT = osrm::contractor::QueryEdge::EdgeData; osrm::storage::Ownership Ownership = (osrm::storage::Ownership)1; osrm::util::StaticGraph<EdgeDataT, Ownership>::Vector<osrm::util::static_graph_details::NodeArrayEntry> = osrm::util::vector_view<osrm::util::static_graph_details::NodeArrayEntry>; osrm::util::StaticGraph<EdgeDataT, Ownership>::Vector<osrm::util::static_graph_details::EdgeArrayEntry<EdgeDataT> > = osrm::util::vector_view<osrm::util::static_graph_details::EdgeArrayEntry<osrm::contractor::QueryEdge::EdgeData> >]: !node_array.empty()
```

of with a segfault if assertions are not enabled.

---

`osrm-routed` should error out early if
- mld is selected and mld data files are missing or
- ch is selected (default) and ch data files are missing

or better: this should be checked in `libosrm` already and not be tied to `osrm-routed`.

**Comment by TheMarex** [comment]
The that was supposed to test this has no implementation for MLD it seems:

https://github.com/Project-OSRM/osrm-backend/blob/master/include/engine/engine.hpp#L146-L202

**Comment by TheMarex** [comment]
Fixed with #4127
